cli/popper/scm.py

```diff
@@ -69,7 +69,7 @@ def get_sha():
     try:
         return repo.git.rev_parse(repo.head.object.hexsha, short=True)
     except ValueError:
-        pu.fail('Needed a single revision.')
+        pu.fail('Repository needs at least one commit\n')
```

Comment:
From the [the docs](https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.rev_parse) looks like the issue is that the given revision could not be parsed. So we might want to write something like:

```python
pu.fail('Could not obtain revision of repository located at {}'.format(get_root_folder()))
```

Reply:
@ivotron so from Stack Trace, I found that [this](https://github.com/gitpython-developers/GitPython/blob/1f66e25c25cde2423917ee18c4704fff83b837d1/git/refs/symbolic.py#L167) is the particular exception that is raised in the GitPython Library .
So I was wondering if this would suffice? 
```
except ValueError as e:
    pu.fail(str(e)+"\n")
``` 
So that error message is `ERROR: Reference at 'refs/heads/master' does not exist` instead of `ERROR: Could not obtain revision of repository located at /home/akash/Desktop/try` generated via 

> pu.fail('Could not obtain revision of repository located at {}'.format(get_root_folder()))

Reply:
Stack Trace for reference is 
```
Traceback (most recent call last):
  File "/usr/local/bin/popper", line 11, in <module>
    load_entry_point('popper==2.0.2.dev0', 'console_scripts', 'popper')()
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/akash/.local/lib/python3.5/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/akash/.local/lib/python3.5/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/popper-2.0.2.dev0-py3.5.egg/popper/commands/cmd_run.py", line 83, in cli
    debug, dry_run, parallel)
  File "/usr/local/lib/python3.5/dist-packages/popper-2.0.2.dev0-py3.5.egg/popper/commands/cmd_run.py", line 89, in run_pipeline
    reuse, parallel)
  File "/usr/local/lib/python3.5/dist-packages/popper-2.0.2.dev0-py3.5.egg/popper/gha.py", line 49, in __init__
    'GITHUB_SHA': scm.get_sha(),
  File "/usr/local/lib/python3.5/dist-packages/popper-2.0.2.dev0-py3.5.egg/popper/scm.py", line 69, in get_sha
    return repo.git.rev_parse(repo.head.object.hexsha, short=True)
  File "/usr/local/lib/python3.5/dist-packages/GitPython-2.1.11-py3.5.egg/git/refs/symbolic.py", line 193, in _get_object
    return Object.new_from_sha(self.repo, hex_to_bin(self.dereference_recursive(self.repo, self.path)))
  File "/usr/local/lib/python3.5/dist-packages/GitPython-2.1.11-py3.5.egg/git/refs/symbolic.py", line 135, in dereference_recursive
    hexsha, ref_path = cls._get_ref_info(repo, ref_path)
  File "/usr/local/lib/python3.5/dist-packages/GitPython-2.1.11-py3.5.egg/git/refs/symbolic.py", line 184, in _get_ref_info
    return cls._get_ref_info_helper(repo, ref_path)
  File "/usr/local/lib/python3.5/dist-packages/GitPython-2.1.11-py3.5.egg/git/refs/symbolic.py", line 167, in _get_ref_info_helper
    raise ValueError("Reference at %r does not exist" % ref_path)
ValueError: Reference at 'refs/heads/master' does not exist
```

Reply:
since not all of popper users are well versed with Git, the message `Reference at 'refs/heads/master' does not exist` might not be friendly to them. Also, printing the path to the repo allows users to give them feedback on whether they are running `popper` against the repo they think they are.

maybe we can check whether `--debug` is enabled, in which case we print the stack trace as well.

Reply:
Updated accordingly